clang-tools-extra/clang-tidy/utils/RenamerClangTidyCheck.cpp

```diff
@@ -367,6 +367,25 @@ class RenamerClangTidyVisitor
     return true;
   }
 
+  bool VisitDesignatedInitExpr(DesignatedInitExpr *Expr) {
+    for (const auto &Designator : Expr->designators()) {
+      if (!Designator.isFieldDesignator())
+        continue;
+      auto *FieldDecl = Designator.getFieldDecl();
```

Comment:
Please do not use `auto` unless type is explicitly stated in same statement or iterator. Should be also `const`.